podman-desktop: 1.16.2 -> 1.17.1

booxter opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

https://github.com/podman-desktop/podman-desktop/releases/tag/v1.17.0
https://github.com/podman-desktop/podman-desktop/releases/tag/v1.17.1

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
podman-desktop: 1.16.2 -> 1.17.1

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>

booxter commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 388848`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

khaneliman commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 388848`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

booxter commented:
@khaneliman I assume x86-64-darwin failure is because of vfkit? (I have a patch for it posted.)

khaneliman commented:
Sorry, forgot I kicked this off

```bash
podman-desktop-x86_64-darwin> dist/extension.js  917.45 kB │ map: 483.38 kB
podman-desktop-x86_64-darwin> ✓ built in 461ms
podman-desktop-x86_64-darwin> Copying files to /private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/extensions/kubectl-cli/builtin/kubectl-cli.cdix
podman-desktop-x86_64-darwin> Zipping files to /private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/extensions/kubectl-cli/kubectl-cli.cdix
podman-desktop-x86_64-darwin>   • electron-builder  version=25.1.8 os=24.3.0
podman-desktop-x86_64-darwin>   • loaded configuration  file=/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/.electron-builder.config.cjs
podman-desktop-x86_64-darwin>   • description is missed in the package.json  appPackageFile=/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/package.json
podman-desktop-x86_64-darwin>   • author is missed in the package.json  appPackageFile=/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/package.json
podman-desktop-x86_64-darwin>   • writing effective config  file=dist/builder-effective-config.yaml
podman-desktop-x86_64-darwin>   • skipped dependencies rebuild  reason=npmRebuild is set to false
podman-desktop-x86_64-darwin>   • packaging       platform=darwin arch=x64 electron=34.3.0 appOutDir=dist/mac
podman-desktop-x86_64-darwin>   • copying Electron  source=/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/electron-dist/Electron.app destination=/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/sour
ce/dist/mac/Electron.app
podman-desktop-x86_64-darwin>   ⨯ spawn EPERM     failedTask=build stackTrace=Error: spawn EPERM
podman-desktop-x86_64-darwin>     at ChildProcess.spawn (node:internal/child_process:420:11)
podman-desktop-x86_64-darwin>     at spawn (node:child_process:753:9)
podman-desktop-x86_64-darwin>     at execFile (node:child_process:346:17)
podman-desktop-x86_64-darwin>     at /private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/builder-util/src/util.ts:99:13
podman-desktop-x86_64-darwin>     at new Promise (<anonymous>)
podman-desktop-x86_64-darwin>     at exec (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/builder-util/src/util.ts:98:10)
podman-desktop-x86_64-darwin>     at getValidIdentities (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/codeSign/macCodeSign.ts:241:11)
podman-desktop-x86_64-darwin>     at _findIdentity (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/codeSign/macCodeSign.ts:276:23)
podman-desktop-x86_64-darwin>     at findIdentity (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/codeSign/macCodeSign.ts:330:14)
podman-desktop-x86_64-darwin>     at MacPackager.sign (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/macPackager.ts:246:36)
podman-desktop-x86_64-darwin>     at /private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/macPackager.ts:498:11
podman-desktop-x86_64-darwin> From previous event:
podman-desktop-x86_64-darwin>     at processImmediate (node:internal/timers:491:21)
podman-desktop-x86_64-darwin> From previous event:
podman-desktop-x86_64-darwin>     at readDirectoryAndSign (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/macPackager.ts:496:29)
podman-desktop-x86_64-darwin>     at MacPackager.signApp (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/macPackager.ts:506:11)
podman-desktop-x86_64-darwin>     at MacPackager.doSignAfterPack (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/platformPackager.ts:346:21)
podman-desktop-x86_64-darwin>     at MacPackager.doPack (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/platformPackager.ts:331:7)
podman-desktop-x86_64-darwin>     at MacPackager.pack (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/macPackager.ts:215:9)
podman-desktop-x86_64-darwin>     at Packager.doBuild (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/packager.ts:459:9)
podman-desktop-x86_64-darwin>     at executeFinally (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/builder-util/src/promise.ts:12:14)
podman-desktop-x86_64-darwin>     at Packager.build (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/app-builder-lib/src/packager.ts:393:31)
podman-desktop-x86_64-darwin>     at executeFinally (/private/tmp/nix-build-podman-desktop-1.17.1.drv-0/source/node_modules/builder-util/src/promise.ts:12:14)

booxter commented:
@khaneliman interesting (I can't reproduce it on my macbook m3 pro). (Also tried with `--argstr sandbox relaxed`.) Are you able to consistently hit it with 1.71.1? (What about 1.16.2?)

I may take a look on the community builder later. Maybe I'll be able to trigger it there.

booxter commented:
This is something about signing attempt in Electron. The message right after your last successful message for me (in my successful build) is

```
skipped macOS application code signing  reason=cannot find valid "Developer ID Application" identity or custom non-Apple code signing certificate, it could cause some undefined behaviour, e.g. macOS localized description not visible, see https://electron.build/code-signing allIdentities=
```

(and then it also lists some signing identities registered on this machine)

khaneliman commented:
> @khaneliman interesting (I can't reproduce it on my macbook m3 pro). (Also tried with `--argstr sandbox relaxed`.) Are you able to consistently hit it with 1.71.1? (What about 1.16.2?)
> 
> I may take a look on the community builder later. Maybe I'll be able to trigger it there.

This was with the community builder. I am gonna try with my macbook.

khaneliman commented:
Yeah, build worked fine on my macbook, too.

booxter commented:
Here's the function that fails:

```
async function getValidIdentities(keychain) {
    function addKeychain(args) {
        if (keychain != null) {
            args.push(keychain);
        }
        return args;
    }
    let result = exports.findIdentityRawResult;
    if (result == null || keychain != null) {
        // https://github.com/electron-userland/electron-builder/issues/481
        // https://github.com/electron-userland/electron-builder/issues/535
        result = Promise.all([
            (0, builder_util_1.exec)("/usr/bin/security", addKeychain(["find-identity", "-v"])).then(it => it
                .trim()
                .split("\n")
                .filter(it => {
                for (const prefix of exports.appleCertificatePrefixes) {
                    if (it.includes(prefix)) {
                        return true;
                    }
                }
                return false;
            })),
            (0, builder_util_1.exec)("/usr/bin/security", addKeychain(["find-identity", "-v", "-p", "codesigning"])).then(it => it.trim().split("\n")),
        ]).then(it => {
            const array = it[0]
                .concat(it[1])
                .filter(it => !it.includes("(Missing required extension)") && !it.includes("valid identities found") && !it.includes("iPhone ") && !it.includes("com.apple.idms.appleid.prd."))
                // remove 1)
                .map(it => it.substring(it.indexOf(")") + 1).trim());
            return Array.from(new Set(array));
        });
        if (keychain == null) {
            exports.findIdentityRawResult = result;
        }
    }
    return result;
}
```

So it fails on `exec`. So it `/usr/bin/security` and fails with `EPERM`.

booxter commented:
OK I know what's going on. It has nothing to do with this version bump. I will send a fix in a second.

khaneliman commented:
> OK I know what's going on. It has nothing to do with this version bump. I will send a fix in a second.

You're right, I just tested against the community builder on master.

booxter commented:
I'll send a separate PR.

khaneliman commented:
Appreciate it, thanks again

booxter commented:
EPERM fix here: https://github.com/NixOS/nixpkgs/pull/388871
